### PR TITLE
fix(ds): File extension detection misidentifies types

### LINKED ISSUE #3181: V3.0.0 File extension detection misidentifies types
**Describe the bug**

Some datasets are misidentified as file types they are not. E.g 'USERID.A.OUTPUT' being detected as '.ASM'. This leads to VS Code setting the language mode according to the incorrect file type.

**To Reproduce**

1. Open a dataset with a single character qualifier, but without an obvious type identifier, e.g USERID.A.OUTPUT, USERID.J.TEST
2. File type will be appended to dataset, e.g "USERID.A.OUTPUT" becomes "USERID.A.OUTPUT.asm", "USERID.J.TEST" becomes "USERID.J.TEST.jcl".

**Expected behavior**

File extensions would only be appended to obviously named data sets, e.g "USERID.SOMETHING.**JCL**" or "USERID.SOMETHING.**COBOL**". 

**Desktop (please complete the following information):**

- OS: Windows 10
- Zowe Explorer Version: v3.0.0

**Additional context**

I **think** this is due to changes to the DatasetUtils.ts file in [eb06448](https://github.com/zowe/zowe-explorer-vscode/commit/eb06448efd0fd8c5e55e42b7de69fddb4d2f3884). The getExtension function is checking if any qualifier is present in each 'match' instead of in the 'matches' array. Before, only qualifiers such as 'JCL' or 'COBOL' would be picked up, but now if a qualifier contains a subset of either, they would be considered.

Additionally, in previous versions, VS Codes "files.associations" setting could be used to allow users to finetune what datasets should be considered what types, though with changes in v3.0.0, this seems to not work anymore for members (Though this may be a separate issue I suppose).

Apologies if I'm missing anything, I don't frequent github / projects, so I hope the above is sufficient context.

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Fixes #3181 

## Release Notes

Milestone: 3.0.1

Changelog: Fixed an issue where automatic file extension detection identified file types incorrectly

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by traeok — DISMISSED** [review]
I know that this is still in draft, but the code and test change LGTM.
Thanks for quickly addressing this oversight

**Review by traeok — DISMISSED** [review]

File: packages/zowe-explorer/src/trees/dataset/DatasetUtils.ts [hunk 1]
```diff
@@ -75,14 +75,8 @@ export class DatasetUtils {
...
             for (const [ext, matches] of DS_EXTENSION_MAP.entries()) {
-                for (const match of matches) {
-                    if (match instanceof RegExp) {
-                        if (match.test(split[i])) {
-                            return ext;
-                        }
-                    } else if (match.includes(split[i])) {
-                        return ext;
-                    }
+                if (matches.some((match) => (match instanceof RegExp ? match.test(split[i]) : match === split[i]))) {
```
Comment (zFernand0) [inline packages/zowe-explorer/src/trees/dataset/DatasetUtils.ts:78]: thanks for rewritting 🙏 
Not sure if this makes it more or less readable, but I kind of prefer it this way (less lines of code) 😅 

And I guess ther diff here is ``` `.incluides` -> `===` ```

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @t1m0thyj

**Review by traeok — APPROVED** [review]

**Pull request merged**